### ISSUE TITLE
Add status command

### ISSUE THREAD

**Issue by Reinmar** [cksource/mgit2#2]
Should pretty-print the statuses of all repos.

We need to figure out which information are the most important and a nice way to print them (some sort of a table?).

**Comment by Reinmar** [comment]
I'm missing this command quite badly. I'm using `mgit exec 'git st'` but it's hard to read when you have 20 repos.

I'd like this command to print a table with the following info:

1. package name,
2. branch name || detached,
3. hash,
4. short status about changes – modified files, files added to the indes, new files,
5. info: ahead/behind/on branch.

How [nodeprompt](https://github.com/oleq/nodeprompt) does it is really good. All important information is present and proper colors are used to make it even easier to read.

**Comment by oleq** [comment]
These are the command that I use when working with branch constellations:
```
mgit exec 'git status | grep "not staged" && git st'
```
```
mgit exec 'git status | grep "not staged" && git diff'
```
```
mgit exec 'git status | grep "On branch t/" && echo $PWD'
```

**Comment by Reinmar** [comment]
I wonder which of these we could:

* include in `mgit st` options,
* for which we could've some configurable shortcuts in `mgit exec`,
* and which deserve their new commands.

I'd say that:

## 1.

```
mgit exec 'git status | grep "not staged" && git st'
```

Could become:

* `mgit st --modified` – prints a table only repos with any changes (includes not staged and untracked files) .
* `mgit st --details` – prints a table but with a column called "Details" that would contain a list of untracked and modified files.

## 2.

```
mgit exec 'git status | grep "not staged" && git diff'
```

Could become `mgit diff`.

The `--modified` command is not needed here because by default we should only print repos where there were any changes.

The important thing here is that we should not lose colors. AFAIK the way we execute commands on multiple repos now means that we lose the colors and we'd need to completely change that bit of the code. Since this method proved to be really stable, I wouldn't like that. So, if there's no other option, that command would need to work like this:

1. retrieve statuses of all repos using the concurrent command execution,
2. check which changed and get diffs for only this repo without involving concurrency.

After all, usually you'll have any changes in just a few repos.

However, I'd really check if we can't preserve colors without changing the execution method.

## 3.

```
mgit exec 'git status | grep "On branch t/" && echo $PWD'
```

This will be really well visible in `mgit st`. Repos which are not on `master` or `major` could be marked in some way – maybe slightly highlighted?

@oleq, WDYT?

**Comment by pomek** [comment]
Status of the `mgit status` command:

![image](https://user-images.githubusercontent.com/2270764/28871104-5915289e-7783-11e7-8d06-9eac6d7d96ab.png)

The status command is available as `st`.

A pink row means – branch in the package is other than defined in `mgit.json`. Other symbols are explained in the Legend.

Thanks @oleq for help to polish the table.

**Comment by pomek** [comment]
Status of the `mgit diff` command:

* For `diff` command we don't display the counter in repository bar:

![image](https://user-images.githubusercontent.com/2270764/28918744-eac70484-784a-11e7-814b-dde17df24bd6.png)

* By default we display changes (not staged) from all repositories:

![image](https://user-images.githubusercontent.com/2270764/28918716-c6f90002-784a-11e7-95ae-8d08c47c5427.png)

* But we can restrict the scope [(already available)](https://github.com/cksource/mgit2/releases/tag/v0.6.0):

![image](https://user-images.githubusercontent.com/2270764/28918808-256489b8-784b-11e7-9135-bcaba56ae3f2.png)

* even more:

![image](https://user-images.githubusercontent.com/2270764/28918827-3a12d860-784b-11e7-8b29-0dd666173de5.png)

* We can pass options directly to the diff command:

![image](https://user-images.githubusercontent.com/2270764/28918876-67083770-784b-11e7-8f73-bcb5f68f90e3.png)

* These options can be defined even if some options for mgit was set:

![image](https://user-images.githubusercontent.com/2270764/28918983-ce7edf30-784b-11e7-8ad8-fc78b26607ed.png)

* When there are no changes, nothing will be displayed:

![image](https://user-images.githubusercontent.com/2270764/28919008-e8c6c20e-784b-11e7-8130-76c827658613.png)

* but if we add changes to staged…

![image](https://user-images.githubusercontent.com/2270764/28919045-0d7a15c4-784c-11e7-88f8-d55ee3c61aab.png)

So basically, we can use `mgit diff` as an alias of `mgit exec 'git diff'` what was the main goal for this command.

**Comment by pomek** [comment]
After all, I executed `mgit st` which prints:

![image](https://user-images.githubusercontent.com/2270764/28919231-e0194b94-784c-11e7-913d-a6fc57a1fc9e.png)
